lib/membrane_audio_mixer_bin.ex

```diff
@@ -0,0 +1,212 @@
+defmodule Membrane.AudioMixerBin do
+  @moduledoc """
+  Bin element distributing a mixing job between multiple `Membrane.AudioMixer` elements.
+
+  A tree of AudioMixers is created according to `max_node_degree` parameter:
+  - if number of input tracks is smaller than `max_node_degree`, only one AudioMixer element is created for the entire job
+  - if there are more input tracks than `max_node_degree`, there are created enough mixers so that each mixer has at most
+  `max_node_degree` inputs - outputs from those mixers are then mixed again following the same rules -
+  another level of mixers is created having enough mixers so that each mixer on this level has at most
+  `max_node_degree` inputs (those are now the outputs of the previous level mixers).
+  Levels are created until only one mixer in the level is needed - output from this mixer is the final mixed track.
+
+  Bin allows for specyfiyng options for `Membrane.AudioMixer`, which are applied for all AudioMixers.
+
+  Recommended to use in case of mixing jobs with many inputs.
+  """
+
+  use Membrane.Bin
+  use Bunch
+
+  require Membrane.Logger
+
+  alias Membrane.{Pad, ParentSpec, AudioMixer}
+  alias Membrane.Caps.Audio.Raw
+  alias Membrane.Caps.Matcher
+
+  @supported_caps {Raw,
+                   format: Matcher.one_of([:s8, :s16le, :s16be, :s24le, :s24be, :s32le, :s32be])}
+
+  def_options max_node_degree: [
+                type: :int,
+                description: """
+                Maximum number of inputs to a single mixer in the mixres tree. Must be at least 2.
+                """,
+                default: 10
+              ],
+              inputs: [
```

Comment:
I don't think passing the number of inputs will be convenient for the user. Can we at least make it optional? In cases where all pads are linked before getting to prepared, it's not needed